Lib/test/_test_multiprocessing.py

```diff
@@ -269,6 +269,62 @@ def _test(cls, q, *args, **kwds):
             q.put(bytes(current.authkey))
             q.put(current.pid)
 
+    def test_parent_process_attributes(self):
+        if self.TYPE == "threads":
+            self.skipTest('test not appropriate for {}'.format(self.TYPE))
+        from multiprocessing.process import current_process
+        rconn, wconn = self.Pipe(duplex=False)
+        p = self.Process(target=self._test_send_parent_process, args=(wconn,))
+        p.start()
+        p.join()
+        parent_process = rconn.recv()
+        assert parent_process.pid == current_process().pid
+        assert parent_process.name == current_process().name
+
+    @classmethod
+    def _test_send_parent_process(cls, wconn):
+        from multiprocessing.process import parent_process
+        wconn.send(parent_process())
+
+    def test_parent_process(self):
+        if self.TYPE == "threads":
+            self.skipTest('test not appropriate for {}'.format(self.TYPE))
+
+        # Launch a child process. Make it launch a grandchild process. Kill the
+        # child process and make sure that the grandchild notices the death of
+        # its parent (a.k.a the child process).
+        rconn, wconn = self.Pipe(duplex=False)
+        p = self.Process(
+            target=self._test_create_grandchild_process, args=(wconn, ))
+        p.start()
+        time.sleep(1)
```

Comment:
This is unsafe as the starting time of the grandchild process might exceed 1s.
I would rather have something where you wait for the grandchild to notify that it is alive, then kill the parent. You could read on `rconn` a first time to get the first `"alive"`, then kill the `child` and read the status again which should be `"not alive"`.